File: app_abarrotes/lib/models/orden_compra.dart
class OrdenCompra {
  final int id;
  final String codigo;
  final int proveedorId;
  final String? fechaEmision;
  final String? fechaEntregaEstimada;
  final String estado;
  final String moneda;
  final String? observaciones;
  final String? proveedorNombre;

  OrdenCompra({
    required this.id,
    required this.codigo,
    required this.proveedorId,
    this.fechaEmision,
    this.fechaEntregaEstimada,
    this.estado = 'pendiente',
    this.moneda = 'PEN',
    this.observaciones,
    this.proveedorNombre,
  });

  factory OrdenCompra.fromJson(Map<String, dynamic> json) {
    return OrdenCompra(
      id: json['id'],
      codigo: json['codigo'],
      proveedorId: json['proveedor_id'],
      fechaEmision: json['fecha_emision'],
      fechaEntregaEstimada: json['fecha_entrega_estimada'],
      estado: json['estado'] ?? 'pendiente',
      moneda: json['moneda'] ?? 'PEN',
      observaciones: json['observaciones'],
      proveedorNombre: json['proveedor']?['nombre'],
    );
  }

  Map<String, dynamic> toJson() => {
    'codigo': codigo,
    'proveedor_id': proveedorId,
    'fecha_emision': fechaEmision,
    'fecha_entrega_estimada': fechaEntregaEstimada,
    'estado': estado,
    'moneda': moneda,
    'observaciones': observaciones,
  };
}

File: backend/app/Models/OrdenCompra.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class OrdenCompra extends Model
{
    protected $table = 'ordenes_compra';

    protected $fillable = [
        'codigo',
        'proveedor_id',
        'solicitud_id',
        'fecha_emision',
        'fecha_entrega_estimada',
        'estado',
        'usuario_crea_id',
        'observaciones',
        'condicion_pago',
        'moneda',
        'tipo_cambio',
    ];

    protected function casts(): array
    {
        return [
            'fecha_emision' => 'datetime',
            'fecha_entrega_estimada' => 'datetime',
            'tipo_cambio' => 'decimal:4',
        ];
    }

    public function proveedor()
    {
        return $this->belongsTo(Proveedor::class);
    }

    public function solicitud()
    {
        return $this->belongsTo(SolicitudCompra::class);
    }

    public function usuarioCrea()
    {
        return $this->belongsTo(User::class, 'usuario_crea_id');
    }

    public function detalles()
    {
        return $this->hasMany(OrdenCompraDetalle::class);
    }

    public function recepciones()
    {
        return $this->hasMany(RecepcionCompra::class);
    }

    /** Compras generadas a partir de esta orden. */
    public function compras()
    {
        return $this->hasMany(Compra::class);
    }
}
